diff --git a/src/gui.c b/src/gui.c
index b20e91c..5695689 100644
--- a/src/gui.c
+++ b/src/gui.c
@@ -54,7 +54,7 @@
 #define MEMORY_ORIGINX (TRACE_ORIGINX) + (TRACE_WIDTH)
 #define MEMORY_ORIGINY (MONITOR_ORIGINY)

-#define CYCLES_SKIP 50
+#define CYCLES_SKIP 50000

 uint8_t io_supports_paint;

@@ -134,6 +134,9 @@ void finish_gui() {
 }

 void trace_emu(char *msg) {
+    if (input_cycle_skip) {
+      return;
+    }
     wprintw(wnd_trace_content, msg);
     wrefresh(wnd_trace_content);
 }
@@ -142,6 +145,14 @@ void update_gui(cpu *m) {
   int read;
   bool keep_going = false;

+  if (m->clock_mode == CLOCK_SPRINT && input_cycle_skip < CYCLES_SKIP)
+    {
+      input_cycle_skip++;
+      return;
+    } else {
+      input_cycle_skip=0;
+    }
+
   do {

     // update LCD contents
@@ -210,11 +221,8 @@ void update_gui(cpu *m) {
     }
     wrefresh(wnd_portmon_content);

-    if (m->clock_mode == CLOCK_SPRINT && input_cycle_skip < CYCLES_SKIP)
-    {
-      input_cycle_skip++;
-    } else {
-      input_cycle_skip=0;
+    if (true)
+    {

       m->k->key_up=false;
       m->k->key_down=false;
diff --git a/src/lcd.c b/src/lcd.c
index cb15637..dfb3190 100644
--- a/src/lcd.c
+++ b/src/lcd.c
@@ -14,7 +14,7 @@ lcd * new_lcd() {
   l->function = 0;
   l->entry_mode = 0;
   l->display_mode = 0;
-  l->data = 0xff;
+  l->data = 0x7f;
   l->cursor = 0;
   memset(l->ddram, 0x00, LCD_MEM_SIZE);
   return l;
